proto/asn1-pdu/asn1_pdu_to_der.cc

```diff
@@ -0,0 +1,159 @@
+#include "asn1_pdu_to_der.h"
+
+namespace asn1_pdu {
+
+// The maximum level of recursion allowed. Values greater than this will just
+// fail.
+static constexpr size_t kRecursionLimit = 200;
+
+// Signal that the recursion limit has been exceeded by a
+// message or value and to abort further processing.
+static bool kErrRecursionLimitReached = false;
```

Comment:
Sorry I had missed this when you changed to a bool and treated it as a sentinel value, rather than a return value.

This should be a class member, like depth_. The original design had this as a static constant (like `#define` but cleaner), but your new design has it as a static global, which we [discourage](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables)

For example, if you had two instances of this fuzzer running (... not like you normally would), they'd share the same state for tracking recursion depth. Making this part of the class should be fine here, as noted in the `.h` file.